src/transformations.rs

```diff
@@ -368,14 +397,32 @@ impl Closurizable for RichTerm {
     /// Generate a fresh variable, bind it to the corresponding closure `(t,with_env)` in `env`,
     /// and return this variable as a fresh term.
     fn closurize(self, env: &mut Environment, with_env: Environment) -> RichTerm {
+        // If the term is already a generated variable (that is, introduced by the share normal
+        // form transformation), we don't have to create a useless intermediate closure. We just
+        // transfer the original thunk to the new environment. This is not only an optimization:
+        // this is relied upon by recursive record merging when computing the fixpoint. Change
+        // carefully.
```

Comment:
“Change carefully” is a bit terse :slightly_smiling_face: . What are the invariants/properties that recursive record merging depends on?

Reply:
Fair enough, will make more explicit.